### ISSUE TITLE
Socket Window WordCounter example error

### ISSUE THREAD

**Issue by vadoverde** [docker-flink/docker-flink#45]
I've been trying to solve this issue with the SocketWindowWordCounter.jar example with no luck. I ran the image local, started the cluster with ./bin/start-cluster.sh, ran nc -l 9000 on a second shell attached to the container, then submitted the example with ./bin/flink run examples/streaming/SocketWindowWordCount.jar. The job hangs for 2 - 3 seconds before throwing this exception:

root@4689051a20fe:/opt/flink# ./bin/flink run examples/streaming/SocketWindowWordCount.jar --port 9000
Starting execution of program

------------------------------------------------------------
 The program finished with the following exception:

org.apache.flink.client.program.ProgramInvocationException: java.net.ConnectException: Connection refused (Connection refused)
        at org.apache.flink.client.program.rest.RestClusterClient.submitJob(RestClusterClient.java:264)
        at org.apache.flink.client.program.ClusterClient.run(ClusterClient.java:464)
        at org.apache.flink.streaming.api.environment.StreamContextEnvironment.execute(StreamContextEnvironment.java:66)
        at org.apache.flink.streaming.examples.socket.SocketWindowWordCount.main(SocketWindowWordCount.java:92)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.apache.flink.client.program.PackagedProgram.callMainMethod(PackagedProgram.java:528)
        at org.apache.flink.client.program.PackagedProgram.invokeInteractiveModeForExecution(PackagedProgram.java:420)
        at org.apache.flink.client.program.ClusterClient.run(ClusterClient.java:404)
        at org.apache.flink.client.cli.CliFrontend.executeProgram(CliFrontend.java:785)
        at org.apache.flink.client.cli.CliFrontend.runProgram(CliFrontend.java:279)
        at org.apache.flink.client.cli.CliFrontend.run(CliFrontend.java:214)
        at org.apache.flink.client.cli.CliFrontend.parseParameters(CliFrontend.java:1025)
        at org.apache.flink.client.cli.CliFrontend.lambda$main$9(CliFrontend.java:1101)
        at org.apache.flink.runtime.security.NoOpSecurityContext.runSecured(NoOpSecurityContext.java:30)
        at org.apache.flink.client.cli.CliFrontend.main(CliFrontend.java:1101)
Caused by: java.net.ConnectException: Connection refused (Connection refused)
        at java.net.PlainSocketImpl.socketConnect(Native Method)
        at java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:350)
        at java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:206)
        at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:188)
        at java.net.SocksSocketImpl.connect(SocksSocketImpl.java:392)
        at java.net.Socket.connect(Socket.java:589)
        at org.apache.flink.streaming.api.functions.source.SocketTextStreamFunction.run(SocketTextStreamFunction.java:96)
        at org.apache.flink.streaming.api.operators.StreamSource.run(StreamSource.java:87)
        at org.apache.flink.streaming.api.operators.StreamSource.run(StreamSource.java:56)
        at org.apache.flink.streaming.runtime.tasks.SourceStreamTask.run(SourceStreamTask.java:99)
        at org.apache.flink.streaming.runtime.tasks.StreamTask.invoke(StreamTask.java:306)
        at org.apache.flink.runtime.taskmanager.Task.run(Task.java:703)
        at java.lang.Thread.run(Thread.java:748)

---
I could run some of the other streaming examples and fetch the results from the task manager's .out file. I tried this on both docker for windows and docker on mac and keep getting the same issue. 

Am I missing something here?

**Comment by patricklucas** [comment]
tl;dr: use `nc -l -p 9000` instead of `nc -l 9000`.

Debian ships with two versions of netcat, `netcat-openbsd` and `netcat-traditional`, and when you just `apt-get install netcat` you get `netcat-traditional`.

You can demonstrate this by first running `apt-get install net-tools netcat-openbsd netcat-traditional`, then:
* `nc.traditional -l 9000`
* `netstat -ptunl`
* Note that there is an `nc` process listening, but not on port 9000
* `nc.traditional -l -p 9000`
* `netstat -ptunl`
* It's listening now on :9000
* `nc.openbsd -l 9000`
* Also listens on :9000

Hope this helps!

**Comment by vadoverde** [comment]
That did it! Thank you for the quick help Patrick...
